### PR TITLE
Add tesseract_scene_graph_example

### PR DESCRIPTION
**Pull request by mpowelson**
This example initializes 2 robots from a URDF. It then reattaches one of the robots to the end effector of the other robot. Here are some screenshots. Attached are the resulting .dot files

Before:
Screenshot from 2020-02-10 11-05-33

After:
Screenshot from 2020-02-10 11-05-52

[scene_graph_example.dot.txt](https://github.com/ros-industrial-consortium/tesseract/files/4182097/scene_graph_example.dot.txt)
[scene_graph_example_moveLink.dot.txt](https://github.com/ros-industrial-consortium/tesseract/files/4182098/scene_graph_example_moveLink.dot.txt)

### CODE REVIEW

File: tesseract_ros/tesseract_examples/tesseract_ros_examples/include/tesseract_ros_examples/scene_graph_example.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/**
+ * @file scene_graph_example.h
+ * @brief This example initializes 2 robots from a URDF. It then reattaches one of the robots to the end effector of the
+ * other robot
+ *
+ * @author Levi Armstrong
```
Comment (Levi-Armstrong) [inline tesseract_ros/tesseract_examples/tesseract_ros_examples/include/tesseract_ros_examples/scene_graph_example.h]: Update date, author and copyright date.

File: tesseract_ros/tesseract_examples/tesseract_ros_examples/src/scene_graph_example.cpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
+/**
+ * @file scene_graph_example.cpp
+ * @brief scene_graph_example implementation
+ *
+ * @author Levi Armstrong
+ * @date July 22, 2019
```
Comment (Levi-Armstrong) [inline tesseract_ros/tesseract_examples/tesseract_ros_examples/src/scene_graph_example.cpp]: Update author, date and copyright date.

File: tesseract_ros/tesseract_examples/tesseract_ros_examples/src/scene_graph_example_node.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/**
+ * @file scene_graph_example_node.cpp
+ * @brief Demonstrates manipulating the scene graph with a robot picking up another robot
+ *
+ * @date Feb 7, 2020
```
Comment (Levi-Armstrong) [inline tesseract_ros/tesseract_examples/tesseract_ros_examples/src/scene_graph_example_node.cpp]: Missing Author and updated copyright date.

File: tesseract_ros/tesseract_examples/tesseract_ros_examples/test/scene_graph_example_unit.cpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#include <tesseract_common/macros.h>
```
Comment (Levi-Armstrong) [inline tesseract_ros/tesseract_examples/tesseract_ros_examples/test/scene_graph_example_unit.cpp]: Missing license.

**Comment by mpowelson** [comment]
@Levi-Armstrong I updated the documentation as requested.

File: tesseract_ros/tesseract_examples/tesseract_ros_examples/src/scene_graph_example.cpp [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  // Attach the iiwa to the end of the ABB using moveLink. The link to be moved is inferred to be the given Joint's
+  // child
+  tesseract_scene_graph::Joint new_joint("to_iiwa_mount");
+  new_joint.parent_link_name = "tool0";
+  new_joint.child_link_name = "iiwa_mount";
+  new_joint.type = tesseract_scene_graph::JointType::FIXED;
+  new_joint.parent_to_joint_origin_transform = Eigen::Isometry3d::Identity();
+  new_joint.parent_to_joint_origin_transform.rotate(Eigen::AngleAxisd(-M_PI / 2, Eigen::Vector3d(0, 1, 0)));
+  new_joint.parent_to_joint_origin_transform.translate(Eigen::Vector3d(0.15, 0.0, 0.0));
+  environment_monitor_->getTesseract()->getEnvironment()->moveLink(new_joint);
```
Comment (Levi-Armstrong) [inline tesseract_ros/tesseract_examples/tesseract_ros_examples/src/scene_graph_example.cpp]: Must now use `std::move(new_joint)`

Comment (mpowelson) [reply tesseract_ros/tesseract_examples/tesseract_ros_examples/src/scene_graph_example.cpp]: Done.

**Pull request merged**